internal/testing/cluster.go

```diff
@@ -163,81 +203,155 @@ func randomPassword() string {
-		return api.AutopilotConfig{}, 0, err
-	}
-	return ap.Config, ap.CurrentPeriod, nil
+	c.tt.OK(err)
+	return ap.Config, ap.CurrentPeriod
 }
 
 // UpdateAutopilotConfig updates the cluster's autopilot with given config.
-func (c *TestCluster) UpdateAutopilotConfig(ctx context.Context, cfg api.AutopilotConfig) error {
-	return c.Bus.UpdateAutopilot(context.Background(), api.Autopilot{
+func (c *TestCluster) UpdateAutopilotConfig(ctx context.Context, cfg api.AutopilotConfig) {
+	c.tt.Helper()
+	c.tt.OK(c.Bus.UpdateAutopilot(context.Background(), api.Autopilot{
 		ID:     c.apID,
 		Config: cfg,
-	})
+	}))
 }
 
-// newTestCluster creates a new cluster without hosts with a funded bus.
-func newTestCluster(dir string, logger *zap.Logger) (*TestCluster, error) {
-	wk := types.GeneratePrivateKey()
-	return newTestClusterWithFunding(dir, "", true, wk, logger)
+type testClusterOptions struct {
+	dbName        string
+	dir           string
+	funding       *bool
+	hosts         int
+	logger        *zap.Logger
+	uploadPacking bool
+	walletKey     *types.PrivateKey
+
+	autopilotCfg      *node.AutopilotConfig
+	autopilotSettings *api.AutopilotConfig
+	busCfg            *node.BusConfig
+	workerCfg         *config.Worker
 }
 
-// newTestClusterWithFunding creates a new cluster without hosts that is funded
-// by mining multiple blocks if 'funding' is set.
-func newTestClusterWithFunding(dir, dbName string, funding bool, wk types.PrivateKey, logger *zap.Logger) (*TestCluster, error) {
-	return newTestClusterCustom(dir, dbName, funding, wk, testBusCfg(), testWorkerCfg(), testApCfg(), logger)
+// newTestLogger creates a console logger used for testing.
+func newTestLogger() *zap.Logger {
+	return newTestLoggerCustom(zapcore.ErrorLevel)
 }
 
-// newTestClusterCustom creates a customisable cluster.
-func newTestClusterCustom(dir, dbName string, funding bool, wk types.PrivateKey, busCfg node.BusConfig, workerCfg config.Worker, apCfg node.AutopilotConfig, logger *zap.Logger) (*TestCluster, error) {
+// newTestLoggerCustom creates a console logger used for testing and allows
+// passing in the desired log level.
+func newTestLoggerCustom(level zapcore.Level) *zap.Logger {
+	config := zap.NewProductionEncoderConfig()
+	config.EncodeTime = zapcore.RFC3339TimeEncoder
+	config.EncodeLevel = zapcore.CapitalColorLevelEncoder
+	config.StacktraceKey = ""
+	consoleEncoder := zapcore.NewConsoleEncoder(config)
+
+	return zap.New(
+		zapcore.NewCore(consoleEncoder, zapcore.AddSync(os.Stdout), level),
+		zap.AddCaller(),
+		zap.AddStacktrace(level),
+	)
+}
+
+// newTestCluster creates a new cluster without hosts with a funded bus.
+func newTestCluster(t *testing.T, opts testClusterOptions) *TestCluster {
+	t.Helper()
+
+	// Skip any test that requires a cluster when running short tests.
+	if testing.Short() {
+		t.SkipNow()
+	}
+	tt := &TT{t}
+
+	// Apply options.
+	dbName := opts.dbName
+	dir := t.TempDir()
+	if opts.dir != "" {
+		dir = opts.dir
+	}
+	logger := newTestLogger()
```

Comment:
`nit` flipping all of these around makes more sense I think

```go
logger := opts.logger
if logger == nil {
    logger = newTestLogger()
}
```

Not sure, it just feels wasteful to create test loggers and test configs if we're about to overwrite it anyway. Another thing I considered was `opts.fillDefaults(t)`, after which you can be sure all options have a sane value.

Reply:
not sure if I really like it but we could also do something like `opts.fillDefaults(t)`